src/eckit/maths/Qhull.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+
+#pragma once
+
+
+#include <string>
+#include <vector>
+
+#include "eckit/exception/Exceptions.h"
+#include "eckit/maths/ConvexHull.h"
+
+#include "libqhullcpp/Qhull.h"
```

Comment:
I would avoid this "libqhullcpp/Qhull.h" include in this place, so that qhull can stay a private dependency of eckit.

Following approach is what I used in atlas:
```
class Qhull {
...
private:
  // forward declaration
  struct Qhull_;
  std::unique_ptr<Qhull_> qh_;
};
```

and in the `.cc` file:
```
#include "libqhullcpp/Qhull.h"

struct Qhull::Qhull_ : public orgQhull::Qhull {
    using orgQhull::Qhull::Qhull;
};

// Qhull constructor
Qhull::Qhull(.....) {
    qh_ = std::make_unique<Qhull>(......);
    ...
}

Reply:
Yes I wasn't too happy with this, I will adapt a similar solution

Reply:
I've made it private with [commit 1c0c4073ae](https://github.com/ecmwf/eckit/pull/99/commits/1c0c4073ae96c46d173413a63db77ed79caa43f0). I've opted for a naked pointer instead (of course with proper new/delete)

Reply:
Fantastic!